fs/semaphore/sem_open.c

```diff
@@ -87,7 +88,7 @@
  *
  ****************************************************************************/
 
-FAR sem_t *nxsem_open(FAR const char *name, int oflags, ...)
+FAR sem_t *nxsem_open(FAR const char *name, int oflags, FAR int *ecode, ...)
```

Comment:
@zhhyu7 do we need to use variadic for nxsem_open() ? The POSIX equivalent from user space is:

```
       sem_t *sem_open(const char *name, int oflag,
                       mode_t mode, unsigned int value);
```
I'm afraid people could pass `mode` at `ecode` possition.

If we can get rid of variadic I suggest you adding `ecode` at end as last parameter

Reply:
Or we can place it as the first parameter

Reply:
> Or we can place it as the first parameter
done.

Reply:
`nxsem_open` is a system call and should always be used via `sem_open`. The prototype for the system call is fine.

Users use libc, hopefully they don't invoke system calls manually by bypassing it.